File: 1/1.js
 
const fs = require('fs');

// load dataset
flist = fs.readFileSync('1.txt')
	.toString()
	.split('\r\n')
	.map(x => parseInt(x));

// day 1.1
const sum = flist.reduce((a, b) => a + b);
console.log(sum);

// day 1.2
const l = flist.length;
let count = 0;
let fsum = 0;
let fset = new Set();
let dupe = null;

while (true) {
	fsum += flist[count];
	if (fset.has(fsum)) {
		break;
	}
	fset.add(fsum);
	count++;
	if (count >= l) count %= l;
}

File: 1/1.py

def main():

	# load dataset
	with open('1.txt', 'r') as f:
		flist = [int(x) for x in f.readlines()]

	# day 1.1
	print(sum(flist))

	# day 1.2
	count = 0
	fsum = 0
	fset = set({})
	l = len(flist)

	while True:
		fsum += flist[count]
		
		if (fsum in fset):
			print(fsum)
			break
		else:
			fset.add(fsum)
		
		count += 1
		if (count >= l): count %= l


# ~~~~~~~~~~~~~~~~~

if __name__ == "__main__":
	main()